internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -446,11 +449,25 @@ func (r *Reconciler) reconcileMachineDeployments(ctx context.Context, s *scope.S
 	diff := calculateMachineDeploymentDiff(s.Current.MachineDeployments, s.Desired.MachineDeployments)
 
 	// Create MachineDeployments.
-	for _, mdTopologyName := range diff.toCreate {
-		md := s.Desired.MachineDeployments[mdTopologyName]
-		if err := r.createMachineDeployment(ctx, s, md); err != nil {
+	if len(diff.toCreate) > 0 {
+		// In current state we only got the MD list via a cached called.
+		// We double-check here that the MachineDeployment doesn't exist, before
+		// we create a duplicate MD.
+		currentMDTopologyNames, err := r.getCurrentMachineDeployments(ctx, s)
+		if err != nil {
 			return err
 		}
+		for _, mdTopologyName := range diff.toCreate {
+			// Skip the MD creation if the MD already exists.
+			if currentMDTopologyNames.Has(mdTopologyName) {
+				continue
```

Comment:
What about adding a log line here so we can somehow track the incidence of this use case